### ISSUE TITLE
Backpack key kicks the client from the server when switching dimensions.

### ISSUE THREAD

**Issue by maxlowry123** [gr8pefish/IronBackpacks#43]
I know there was an issue like this before but I am still having the issue. I have tried with every version since 1.2.10 up to 1.2.13. The issue says it was fixed in 1.2.8 but it is still happening. When I press 'R' to open the backpack after switching dimensions I get kicked from my server with http://pastebin.com/q2bUZDHP
As the other person said unequipping the backpack and reequipping it fixes it but if I did not have to do that it would be nice.
List of mods: http://pastebin.com/WtwKUrid

I tried deleting the configs on both the client and the server to see if that was the issue and it is still happening.
If it would help I can post the full client log.

**Comment by gr8pefish** [comment]
Dang, I thought I fixed this one. 

I'm having trouble duplicating it myself. Can you, in detail, describe a chain of events that has led/would lead to this crash?

**Comment by maxlowry123** [comment]
I am using XP Teleports to move between dimensions and unless I reequip the backpack the server kicks me when I open the backpack using 'R'. I am not sure if it would help but I can send you a link to the pack and whitelist you on the server and you can see if it happens there.

**Comment by gr8pefish** [comment]
Is it easy for you to tell me if it works for you if you just hop through a nether portal? I will try adding XP teleporters and testing if it is that mod specifically that is breaking it.

**Comment by maxlowry123** [comment]
Nether portals work fine.

**Comment by gr8pefish** [comment]
Okay yup, I think it is that mod that is doing something funky. When I go through the teleporter and try to access the backpack I get kicked (as it isn't there/isn't rendered), but then once I join again I can immediately access it just fine (and it is there rendered on the player again), is that the case with you as well?

**Comment by maxlowry123** [comment]
Yes, the same thing happens to me.

**Comment by gr8pefish** [comment]
Okay so what I am guessing is happening is that that mod is failing to do is call a method named "onDimensionChanged," it just directly teleports the player so my mod doesn't know anything has happened, and, as such, can't correctly update the backpack on the player. When it reloads, vanilla minecraft loads the player and my mod notices that and puts the backpack where it should be (on the player). 

So long story short, this one is not on my end, this one is XPTeleporter's fault. I'll try to contact him, for now it just doesn't work I guess, but there is nothing I can do about it.

**Comment by maxlowry123** [comment]
Ok, thanks. I also noticed that the same thing happens with FTB Utilities teleporting. Both /warp and /home and maybe other ways of teleporting.

**Comment by gr8pefish** [comment]
That would make sense, they are all by the same person, so if there is a mistake in his lib code it would cause multiple errors like the ones you have mentioned. I just talked to him on IRC, he said he'll fix it.

Also, while I have you, any comments about this mod? Any suggestions/improvements/annoyances/etc?

**Comment by maxlowry123** [comment]
One thing that is a bit annoying it when using the nesting upgrade and opening a backpack that is inside. When you do that sometimes one of the items gets picked up in the backpack you just opened with a right click. 
![2016-01-08_18 58 41](https://cloud.githubusercontent.com/assets/5221871/12212573/226d974c-b63a-11e5-896f-6362e56d9753.png)
![2016-01-08_18 58 45](https://cloud.githubusercontent.com/assets/5221871/12212575/27c9c4ae-b63a-11e5-8da7-90bcbfda6e8a.png)

Other than that there is not anything that I can think of right now.

**Comment by gr8pefish** [comment]
Hmm I can't get that one to happen for me, I'm guessing it's inventory tweaks or some other mod, or maybe even server latency. If you can figure out what is happening I can fix it, but it seems minor enough I'm not going to worry about it, because it looks like one of those tricky bugs to nail down.

If you have any other suggestions I'll be here :)

**Comment by maxlowry123** [comment]
That is fine, as I said it is just a bit annoy but if it is not a quick fix then it is not worth the time.
